testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2523,8 +2524,10 @@ public void testRegisterTableWithTableLocation(StorageFormat storageFormat)
         String sparkTableName = sparkTableName(baseTableName);
 
         onSpark().executeQuery(format("CREATE TABLE %s (a INT, b STRING, c BOOLEAN) USING ICEBERG TBLPROPERTIES ('write.format.default' = '%s')", sparkTableName, storageFormat));
+        // Run both inserts on Spark; When an insert on Trino fails and is retried, it creates a second metadata file with same sequence number in the filename,
+        // and register_table would detect this ambiguity and fail.
         onSpark().executeQuery(format("INSERT INTO %s values(1, 'INDIA', true)", sparkTableName));
-        onTrino().executeQuery(format("INSERT INTO %s values(2, 'USA', false)", trinoTableName));
+        onSpark().executeQuery(format("INSERT INTO %s values(2, 'USA', false)", trinoTableName));
```

Comment:
trinoTableName should be sparkTableName. 

Also, just removing this line might be acceptable. Assuming the test purpose is registering a table created by Spark.